### PR TITLE
chore(deps): Migrate to lerna v7

### LINKED ISSUE #779: Lerna Bootstrap no longer supported as of Lerna v7.0
# Bug Report 🐛

As of Lerna version 7, the bootstrap command has been removed by default. Instead, Lerna now relies on npm's workspaces feature to manage dependencies within a monorepo.

## Current Behaviour
```
$ lerna bootstrap

ERR! bootstrap The "bootstrap" command was removed by default in v7, and is no longer maintained.
ERR! bootstrap Learn more about this change at https://lerna.js.org/docs/legacy-package-management
```

## Possible Solution
   - Possibly modify repo to use npm's workspaces?
   - Something else?

### PR DESCRIPTION
**Pull request by martinhalford**
# Closes #779 
Lerna Bootstrap has been deprecated as of Lerna version 7.  

The linking of `packages/*` is now handled by `npm` using `workspaces`.  

### Changes
- Remove `lerna bootstrap`.
- Add `workspaces` to `package.json` - so that `npm` handles package linking.
- Update devDependencies: `"lerna": "^7.3.1",`
- Remove references to `lerna bootstrap` from `DEVELOPERS.md`. 

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Extend the documentation, if necessary
- [x] Merging to `master` from `fork:branchname`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "dayjs": "1.11.8",
     "eslint": "^8.43.0",
     "jsdoc": "^4.0.2",
-    "lerna": "^5.5.2",
+    "lerna": "^7.3.1",
```
[the remark this answers is not in the document]
Comment (martinhalford) [reply package.json:8]: According to the tutorial I followed, this was dependency was updated but not removed.  I assumed Lerna may be used elsewhere and, therefore this reference is needed?
https://www.youtube.com/watch?v=NEFJNz4wx1k

File: package.json [hunk 2]
```diff
@@ -19,18 +19,19 @@
...
+    "packages/*"
+  ],
   "scripts": {
     "coverage": "node ./scripts/coverage.js \"packages/cicero-*\" && nyc report -t coverage --cwd . --report-dir coverage --reporter=lcov && cat ./coverage/lcov.info",
-    "postinstall": "npm run bootstrap",
-    "bootstrap": "lerna bootstrap",
     "pretest": "npm run licchk",
     "test": "lerna exec -- npm run test:cov",
     "repoclean": "lerna clean",
     "licchk": "lerna exec -- npm run licchk"
```
[the remark this answers is not in the document]
Comment (martinhalford) [reply package.json:30]: I'm not sure what the equivalent npm commands are.  I assumed we can keep using Lerna?

**Comment by martinhalford** [comment]
> Let's go all the way and remove the dependency on Lerna.
> 
> Dan did a similar thing for markdown-transform recently, https://github.com/accordproject/markdown-transform/pull/570/files
> 
> Can you also review the GitHub Actions scripts (in `.github/workflows`) too? Especially the build and publish steps.

OK. Let me see what I can do.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinhalford) [reply package.json:8]: Removed

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinhalford) [reply package.json:30]: Removed, if redundant, or replaced.

**Comment by martinhalford** [comment]
> > Let's go all the way and remove the dependency on Lerna.
> > Dan did a similar thing for markdown-transform recently, https://github.com/accordproject/markdown-transform/pull/570/files
> > Can you also review the GitHub Actions scripts (in `.github/workflows`) too? Especially the build and publish steps.
> 
> OK. Let me see what I can do.

@mttrbrts 

I've removed Lerna completely.  However, some stuff has been modified but not tested - most notably `.github/workflows` which were copy/pastes of @dselman's code from Markdown Transform. 

I've also tweaked the `license.config.json` files; and renamed `lerna.json` to `config.json` and tweaked it, as that was still being referenced by `scripts/timestamp.js`.

**Review by mttrbrts — APPROVED** [review]

**Pull request merged**